### PR TITLE
[mt][browser] Prevent `SignalRClientTests` fail with "PlaywrightException: Target closed" 

### LINKED ISSUE #99483: SignalRClientTests fail with "PlaywrightException: Target closed"
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=596073
Build error leg or test failing: Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution
Pull request: https://github.com/dotnet/runtime/pull/99185

## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "Unhandled exception. Microsoft.Playwright.PlaywrightException: Target closed",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=596073
**Error message validated:** `[Unhandled exception. Microsoft.Playwright.PlaywrightException: Target closed`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 3/9/2024 3:41:09 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[604256](https://dev.azure.com/dnceng-public/public/_build/results?buildId=604256)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=604256&view=ms.vss-test-web.build-test-results-tab&runId=14702928&resultId=100164)|dotnet/runtime#99659|
|[603541](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603541)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603541&view=ms.vss-test-web.build-test-results-tab&runId=14676658&resultId=100164)|dotnet/runtime#99703|
|[603487](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603487)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603487&view=ms.vss-test-web.build-test-results-tab&runId=14673526&resultId=100164)|dotnet/runtime#99637|
|[603199](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603199)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603199&view=ms.vss-test-web.build-test-results-tab&runId=14664882&resultId=100164)|dotnet/runtime#99676|
|[603162](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603162)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=603162&view=ms.vss-test-web.build-test-results-tab&runId=14662382&resultId=100166)|dotnet/runtime#99756|
|[602604](https://dev.azure.com/dnceng-public/public/_build/results?buildId=602604)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=602604&view=ms.vss-test-web.build-test-results-tab&runId=14643538&resultId=100164)|dotnet/runtime#99439|
|[601942](https://dev.azure.com/dnceng-public/public/_build/results?buildId=601942)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=601942&view=ms.vss-test-web.build-test-results-tab&runId=14626136&resultId=100164)|dotnet/runtime#99422|
|[601483](https://dev.azure.com/dnceng-public/public/_build/results?buildId=601483)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=601483&view=ms.vss-test-web.build-test-results-tab&runId=14599944&resultId=100164)|dotnet/runtime#98761|
|[599835](https://dev.azure.com/dnceng-public/public/_build/results?buildId=599835)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=599835&view=ms.vss-test-web.build-test-results-tab&runId=14545678&resultId=100001)|dotnet/runtime#99537|
|[599517](https://dev.azure.com/dnceng-public/public/_build/results?buildId=599517)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=599517&view=ms.vss-test-web.build-test-results-tab&runId=14533588&resultId=100166)|dotnet/runtime#99422|
|[598225](https://dev.azure.com/dnceng-public/public/_build/results?buildId=598225)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=598225&view=ms.vss-test-web.build-test-results-tab&runId=14495750&resultId=100164)|dotnet/runtime#99550|
|[597464](https://dev.azure.com/dnceng-public/public/_build/results?buildId=597464)|dotnet/runtime|[Workloads-NoWebcil-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=597464&view=ms.vss-test-web.build-test-results-tab&runId=14476962&resultId=100164)|dotnet/runtime#99524|
|[596647](https://dev.azure.com/dnceng-public/public/_build/results?buildId=596647)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=596647&view=ms.vss-test-web.build-test-results-tab&runId=14445708&resultId=100166)||
|[596073](https://dev.azure.com/dnceng-public/public/_build/results?buildId=596073)|dotnet/runtime|[Workloads-Wasm.Build.Tests.TestAppScenarios.SignalRClientTests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=596073&view=ms.vss-test-web.build-test-results-tab&runId=14425304&resultId=100163)|dotnet/runtime#99185|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|5|14|14|

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/99483.

### CODE REVIEW

**Review by pavelsavara — APPROVED** [review]
please revert eng/testing/scenarios/BuildWasmAppsJobsList.txt

**Comment by ilonatommy** [comment]
> please revert eng/testing/scenarios/BuildWasmAppsJobsList.txt

Will do. But on the last (3rd) run the test hit the problem again, so it's not a fix. Will try with just a delay you proposed.

File: src/mono/wasm/Wasm.Build.Tests/Blazor/BlazorWasmTestBase.cs [hunk 1]
```diff
@@ -209,7 +209,7 @@ public Task BlazorRunForPublishWithWebServer(BlazorRunOptions runOptions)
             modifyBrowserUrl: browserUrl => browserUrl + runOptions.BrowserPath + runOptions.QueryString);
 
         _testOutput.WriteLine("Waiting for page to load");
-        await page.WaitForLoadStateAsync(LoadState.DOMContentLoaded, new () { Timeout = 1 * 60 * 1000 });
+        await page.WaitForLoadStateAsync(LoadState.NetworkIdle, new () { Timeout = 1 * 60 * 1000 });
```
Comment (maraf) [inline src/mono/wasm/Wasm.Build.Tests/Blazor/BlazorWasmTestBase.cs:212]: Could we instead wait on some console message produced by a rendered component?

File: src/mono/wasm/testassets/BlazorHostedApp/BlazorHosted.Client/Pages/Chat.razor [hunk 2]
```diff
@@ -27,7 +27,7 @@
     private List<string> chatMessages = new List<string>();
     private string wrongQueryError = "Query string with parameters 'message' and 'transport' are required";
 
-    protected override void OnAfterRender()
+    protected override void OnAfterRender(bool firstRender)
     {
         Helper.TestOutputWriteLine($"OnAfterRender on CurrentManagedThreadId={Environment.CurrentManagedThreadId}");
```
Comment (maraf) [inline src/mono/wasm/testassets/BlazorHostedApp/BlazorHosted.Client/Pages/Chat.razor:32]: nit: Call base

```suggestion
    protected override void OnAfterRender(bool firstRender)
    {
        base.OnAfterRender(firstRender);

        Helper.TestOutputWriteLine($"OnAfterRender on CurrentManagedThreadId={Environment.CurrentManagedThreadId}");
```

**Review by maraf — APPROVED** [review]

**Pull request merged**